wagtail/admin/forms/pages.py

```diff
@@ -226,3 +226,31 @@ def __init__(self, *args, **kwargs):
             label=_("New parent page"),
             help_text=_("Select a new parent for this page."),
         )
+
+
+class PageChoiceField(forms.ModelChoiceField):
+    def label_from_instance(self, obj):
+        if obj.is_root():
+            return obj.get_admin_display_title()
```

Comment:
We exclude the root page when we are trying to get the ancestor, this leads to the label for the root page itself rendering as blank, so I had to add this.

Or should we prevent the user from choosing the user from choosing the root here as a parent ?